python/meterpreter/meterpreter.py

```diff
@@ -630,6 +640,48 @@ def seek(self, offset, whence=os.SEEK_SET):
     def tell(self):
         raise NotImplementedError()
 
+#@export
+class MeterpreterLog(MeterpreterChannel):
+    def __init__(self, level, size, active):
+        self._buffer = ''
+        self.level = level
+        self.size = size
+        self.active = active
+
+    def close(self):
+        self.active = False
+
+    def eof(self):
+        return False
+
+    def read(self, length):
+        chunk = self._buffer[:length]
+        self._buffer = self._buffer[length:]
+        return chunk
+
+    def write(self, data):
+        self._buffer += data
+        self._buffer = self._buffer[-self.size:]
+
+    def log_debug(self, message):
+        if not self.active or self.level > LOG_LEVEL_DEBUG:
+            return
+        self.write(message)
+
+    def log_info(self, message):
+        if not self.active or self.level > LOG_LEVEL_INFO:
+            return
+        self.write(message)
+
+    def log_error(self, message):
+        if not self.active or self.level > LOG_LEVEL_ERROR:
+            return
+        self.write(message)
+export(MeterpreterLog)
+
+# If debugging is enabled start recording logs to the channel immediately
+log_channel = MeterpreterLog(LOG_LEVEL_INFO, 0x2000, DEBUGGING)
```

Comment:
Should this magic number be extracted to a constant so the value can be patched by the user? 👀 

Is this magic number large enough to capture all of stdapi's default logging by default? 🤔